crates/node/src/indexer/tx_sender.rs

```diff
@@ -201,119 +172,40 @@ async fn observe_tx_result(
         Respond(respond_args) => {
             // Confirm whether the respond call succeeded by checking whether the
             // pending signature request still exists in the contract state
-            let get_pending_request_args: Vec<u8> =
-                serde_json::to_string(&ChainGetPendingSignatureRequestArgs {
-                    request: respond_args.request.clone(),
-                })
-                .unwrap()
-                .into_bytes();
-            let query_response = indexer_state
+            let pending_request_response = indexer_state
                 .view_client
-                .send(
-                    Query {
-                        block_reference: BlockReference::Finality(Finality::Final),
-                        request: QueryRequest::CallFunction {
-                            account_id: indexer_state.mpc_contract_id.clone(),
-                            method_name: "get_pending_request".to_string(),
-                            args: get_pending_request_args.into(),
-                        },
-                    }
-                    .with_span_context(),
-                )
-                .await??;
-            match query_response.kind {
-                QueryResponseKind::CallResult(call_result) => {
-                    let pending_request =
-                        serde_json::from_slice::<Option<YieldIndex>>(&call_result.result)?;
-                    Ok(if pending_request.is_none() {
-                        TransactionStatus::Executed
-                    } else {
-                        TransactionStatus::NotExecuted
-                    })
-                }
-                _ => {
-                    anyhow::bail!("Unexpected result from a view client function call");
-                }
-            }
+                .get_pending_request(&indexer_state.mpc_contract_id, &respond_args.request)
```

Comment:
Wouldn't we expect `view_client.get_pending_request` to return `None` in case the response has been submitted?

Reply:
Yeah this seems like a race condition. However it's unrelated to the PR since the previous logic seems to be equivalent.